perf/debug_warning.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <iostream>
+
+static void
+show_warning_if_debug()
+{
+#ifndef NDEBUG
+	std::cerr << "#######################################################\n"
+		  << "#######################################################\n"
+		  << "#######################################################\n"
+		  << "###                                                 ###\n"
+		  << "###                    WARNING!                     ###\n"
+		  << "###   The performance test is run in debug build!   ###\n"
+		  << "###   Test results are definitely inappropriate!    ###\n"
+		  << "###                                                 ###\n"
+		  << "#######################################################\n"
+		  << "#######################################################\n"
+		  << "#######################################################\n";
+#endif // #ifndef NDEBUG
+}
+
+struct DebugWarning {
+	DebugWarning()
+	{
+		show_warning_if_debug();
+	}
+} debug_warning;
```

Comment:
AFAIC, this object must be `static` and its constructor must be marked `noexcept` (as well as `show_warning_if_debug`).

Reply:
Why?

Reply:
Well, first of all, you must not create non-static objects in header files, I guess that's obvious...

Second, if your constructor throws an exception during an initialization of a static variable, it is impossible to catch, thus your program will be terminated. So you'd better declare your constructor `noexcept` to keep this in mind.

Reply:
Ok, agree, fixed.